src/libraries/BN254.sol

```diff
@@ -62,7 +62,7 @@ library BN254 {
     ///      this is because of the (unknown to us) convention used in the bn254 pairing precompile contract
     ///      "Elements a * i + b of F_p^2 are encoded as two elements of F_p, (a, b)."
     ///      https://github.com/ethereum/EIPs/blob/master/EIPS/eip-197.md#encoding
-    function generatorG2() internal pure returns (G2Point memory) {
+    function generatorG2() external pure returns (G2Point memory) {
```

Comment:
Is making this library external only way to decrease code size?

Reply:
Yeah we are at ~30 bytes remaining

Reply:
I did this one the functions that aren't called multiple times in loops so the gas impact was minimal